### PR TITLE
[Feature] support prepare hidden states use dp

### PR DESCRIPTION
**Pull request by jiapingW**
## Motivation
It supports using DP to generate hidden states offline. Currently, it uses the HF backend because the Sglang backend has a relatively high GPU memory overhead (this will be analyzed and optimized later). A qwen2.5 instance has also been added. Related tests will be added later.
The current design involves splitting the dataset into DP segments, then having the backend process one batch at a time and write it to a single file. Additionally, data is written to a separate folder for every 2,000 records. The part code is from [here](https://github.com/sgl-project/SpecForge/pull/268).

## Checklist

- [✅] Format your code according to the [Code Formatting with Pre-Commit](https://docs.sglang.ai/references/contribution_guide.html#code-formatting-with-pre-commit).
- [ ] Add unit tests as outlined in the [Running Unit Tests](https://docs.sglang.ai/references/contribution_guide.html#running-unit-tests-adding-to-ci).
- [ ] Update documentation / docstrings / example tutorials as needed, according to [Writing Documentation](https://docs.sglang.ai/references/contribution_guide.html#writing-documentation-running-docs-ci).
- [ ] Provide throughput / latency benchmark results and accuracy evaluation results as needed, according to [Benchmark and Profiling](https://docs.sglang.ai/references/benchmark_and_profiling.html) and [Accuracy Results](https://docs.sglang.ai/references/accuracy_evaluation.html).
- [ ] For reviewers: If you haven't made any contributions to this PR and are only assisting with merging the main branch, please remove yourself as a co-author when merging the PR.
- [ ] Please feel free to join our Slack channel at https://sgl-fru7574.slack.com/archives/C09784E3EN6 to discuss your PR.

### CODE REVIEW

**Comment by zyksir** [comment]
I am adding the draft tp feature, we can merge this after this feature. since if we want to enable draft tp, we need to decouple the draft tp and the target tp, which leads to a large code refactor.

**Comment by FrankLeeeee** [comment]
> I am adding the draft tp feature, we can merge this after this feature. since if we want to enable draft tp, we need to decouple the draft tp and the target tp, which leads to a large code refactor.

This PR is not about draft model and i guess yours should not clash with this PR significantly? i happen to fix the hidden states generation in the branch https://github.com/sgl-project/SpecForge/tree/hotfix/prepare-hidden-states but it is alright, I will add some optimization further after this PR. Can we merge @jiapingW 's PR first to fix the broken script? @zyksir

**Comment by jiapingW** [comment]
Sorry, there's a small bug here that I haven't updated: https://github.com/sgl-project/SpecForge/blob/main/scripts/prepare_hidden_states.py#L377. Since I don't have my computer with me now, I'll submit it tonight, or you can use this fix directly. Thank you.
## Origin code
```python 
seq_lengths = eagle3_data.attention_mask.sum(dim=1).tolist()
hidden_states_cpu = eagle3_data.hidden_states.cpu()
aux_hidden_states_cpu = None
if self.enable_aux_hidden_states and hasattr(
    eagle3_data, "aux_hidden_states"
):
    aux_hidden_states_cpu = eagle3_data.aux_hidden_states.cpu()

for i, (current_global_idx, seq_len) in enumerate(
    zip(sample_global_indices, seq_lengths)
):
    data_point = {
        "input_ids": filtered_batch["input_ids"][i].clone(),
        "loss_mask": filtered_batch["loss_mask"][i].clone(),
        "hidden_state": hidden_states_cpu[i, :seq_len, :]
        .clone()
        .unsqueeze(0),
    }
    if aux_hidden_states_cpu is not None:
        data_point["aux_hidden_state"] = (
            aux_hidden_states_cpu[i, :seq_len, :].clone().unsqueeze(0)
        )

    output_file = self._get_file_path(output_path, current_global_idx)
    self._save_tensor_async(data_point, output_file)

del hidden_states_cpu, aux_hidden_states_cpu
total_processed += len(sample_global_indices)
```

## Updated Code
```python
if not hasattr(eagle3_data, "target"):
    raise AttributeError(
        "The result from generate_eagle3_data does not have a 'target' attribute, "
        "which is required for the 'hidden_state' field."
    )
if self.enable_aux_hidden_states and not hasattr(eagle3_data, "hidden_states"):
    raise AttributeError(
        "The result from generate_eagle3_data does not have a 'hidden_states' attribute, "
        "which is required for the 'aux_hidden_state' field."
    )

seq_lengths = eagle3_data.attention_mask.sum(dim=1).tolist()

target_hidden_states_cpu = eagle3_data.target.cpu()

aux_hidden_states_cpu = None
if self.enable_aux_hidden_states:
    aux_hidden_states_cpu = eagle3_data.hidden_states.cpu()

for i, (current_global_idx, seq_len) in enumerate(
    zip(sample_global_indices, seq_lengths)
):
    data_point = {
        "input_ids": filtered_batch["input_ids"][i].clone(),
        "loss_mask": filtered_batch["loss_mask"][i].clone(),
    }

    data_point["hidden_state"] = (
        target_hidden_states_cpu[i, :seq_len, :].clone().unsqueeze(0)
    )

    if aux_hidden_states_cpu is not None:
        data_point["aux_hidden_state"] = (
            aux_hidden_states_cpu[i, :seq_len, :].clone().unsqueeze(0)
        )

    output_file = self._get_file_path(output_path, current_global_idx)
    self._save_tensor_async(data_point, output_file)

del target_hidden_states_cpu
if aux_hidden_states_cpu is not None:
    del aux_hidden_states_cpu

total_processed += len(sample_global_indices)
````
